wandb/sdk/wandb_run.py

```diff
@@ -1959,16 +1997,160 @@ def _on_ready(self) -> None:
         # object is about to be returned to the user, don't let them modify it
         self._freeze()
 
+    def _log_job(self) -> None:
+        artifact = None
+        input_types = TypeRegistry.type_of(self.config.as_dict()).to_json()
+        output_types = TypeRegistry.type_of(self.summary._as_dict()).to_json()
+
+        import pkg_resources
+
+        installed_packages_list = sorted(
+            f"{d.key}=={d.version}" for d in iter(pkg_resources.working_set)
+        )
+
+        for job_creation_function in [
+            self._create_repo_job,
+            self._create_artifact_job,
+            self._create_image_job,
+        ]:
+            artifact = job_creation_function(
+                input_types, output_types, installed_packages_list
+            )
+            if artifact:
+                artifact.wait()
+                if not artifact.metadata:
+                    artifact.metadata["config_defaults"] = self.config.as_dict()
+                    artifact.save()
```

Comment:
Yeah, we can make the default configs for a job version the latest instance of that job version. This was discussed earlier as well. But a note, job creation, at least for now is disallowed in offline mode.

Reply:
Also note, jobs created from the code artifacts require a wait call as well. Since we need the artifact name and index to be single ground truth source for the underlying code artifact

Reply:
Where is that in this PR? maybe it is in the other one? We overcome something similar with logging tables using the `client-artifact-id` ... basically the client makes a big random ID which is associated with the eventual true ID. that client ID can be used for lookups

Reply:
Nice, I'll make that change.

Reply:
Oh wait, since this id is unique, this really can't be done. Since the job artifact name needs to be consistent across runs executing the same code.

Reply:
Nvm I got this no more wait.
1. We create a dict self._code_artifact_info =. {"name": <code artifact name>, "client_id": <code-artifact-client-id>}`
2. use the name to create the job artifact.
3. Use the client id to refer to the actual artifact in the source.
Just tested and it works.